lib/src/port.dart

```diff
@@ -226,8 +226,10 @@ class _SerialPortImpl implements SerialPort {
   }
 
   static List<String> get availablePorts {
+    int rv;
     final out = ffi.allocate<ffi.Pointer<ffi.Pointer<sp_port>>>();
-    Util.call(() => dylib.sp_list_ports(out));
+    Util.call(() => rv = dylib.sp_list_ports(out));
+    if (rv != sp_return.SP_OK) return [];
```

Comment:
The `ffi.free()` calls below need to be executed to avoid leaking resources allocated above.

Reply:
you are right!